File: test/nodejs/saneql_test.js
var duckdb = require('../../duckdb/tools/nodejs');
var assert = require('assert');

describe(`saneql extension`, () => {
    let db;
    let conn;
    before((done) => {
        db = new duckdb.Database(':memory:', {"allow_unsigned_extensions":"true"});
        conn = new duckdb.Connection(db);
        conn.exec(`LOAD '${process.env.SANEQL_EXTENSION_BINARY_PATH}';`, function (err) {
            if (err) throw err;
            done();
        });
        // create table foo (a int); and populate it with some data
        conn.exec(`create table foo (a int);`, function (err) {
            if (err) throw err;
            done();
        });

        conn.exec(`insert into foo values (1), (2), (3);`, function (err) {
            if (err) throw err;
            done();
        });
    });

    it('saneql function should return expected string', function (done) {
        db.all("foo.filter(a = 1)", function (err, res) {
            if (err) throw err;
            assert.deepEqual(res, [{value: 1}]);
            done();
        });
    });
});

File: test/python/saneql_test.py
import duckdb
import os
import pytest

# Get a fresh connection to DuckDB with the saneql extension binary loaded
@pytest.fixture
def duckdb_conn():
    extension_binary = os.getenv('SANEQL_EXTENSION_BINARY_PATH')
    if (extension_binary == ''):
        raise Exception('Please make sure the `SANEQL_EXTENSION_BINARY_PATH` is set to run the python tests')
    conn = duckdb.connect('', config={'allow_unsigned_extensions': 'true'})
    conn.execute(f"load '{extension_binary}'")
    conn.execute(f"create table foo (a int)")
    conn.execute(f"insert into foo values (1), (2), (3), (42)")
    return conn

def test_saneql(duckdb_conn):
    duckdb_conn.execute("foo.filter(a > 2)");
    res = duckdb_conn.fetchall()
    assert(res[0][0] == "1");
